content/docs/user-guide/project-structure/dvcyaml-files.md

```diff
@@ -265,6 +265,7 @@ These include a subset of the fields in `.dvc` file
 | `persist`    | Whether the output file/dir should remain in place during `dvc repro` (`false` by default: outputs are deleted when `dvc repro` starts)                                                                                                                                           |
 | `checkpoint` | (Optional) Set to `true` to let DVC know that this output is associated with [checkpoint experiments](/doc/user-guide/experiment-management/checkpoints). These outputs are reverted to their last cached version at `dvc exp run` and also `persist` during the stage execution. |
 | `desc`       | (Optional) User description for this output. This doesn't affect any DVC operations.                                                                                                                                                                                              |
+| `push`       | Whether or not this file or directory, when previously <abbr>cached</abbr>, is uploaded to remote storage by `dvc push` (`true` by default).                                                                                                                                      |
```

Comment:
> Echoing https://github.com/iterative/dvc/pull/8581#issuecomment-1336274557:

Should we plan to recommend this a lot in Data Pipeline docs? Specifically for intermediate pipeline outputs. Assuming the happy path out there is to push only raw data and final ML model files.

If we don't at least emphasize the possibility, users may realize too late they have pushed a bunch of intermediate output versions and they are pretty difficult to clean up with `dvc gc` ([support example](https://discord.com/channels/485586884165107732/1047778499999576084/1048151267190509600)).

Reply:
I'm not sure that not pushing is the right default behavior, even for intermediate outputs. If the user wants to take advantage of run-cache to not need to re-run stages that have already been reproduced, they still need to push/pull intermediate outs

Reply:
@jorgeorpinel Thinking about it some more, I like the suggestion and think it makes sense as a possible product direction to make it easier to get started with pipelines, so let's brainstorm more on it.